cmd/earthly/main.go

```diff
@@ -810,6 +810,18 @@ func newEarthlyApp(ctx context.Context, console conslogging.ConsoleLogger) *eart
 			Usage:  "Edits your Earthly configuration file",
 			Hidden: true, // Experimental.
 			Action: app.actionConfig,
+			Description: `This command takes a path, and a value and sets it in your configuration file.
+
+As the configuration file is YAML, the key must be a valid key within the file. You can specify sub-keys by using "." to separate levels.
+If the sub-key you wish to use has a "." in it, you can quote that subsection, like this: git."github.com".
+
+Values must be valid YAML, and also be deserializable into the key you wish to assign them to.
+This means you can set higher level objects using a compact style, or single values with simple values.
+
+Only one key/value can be set per invocation.
+
```

Comment:
Would it be possible to add an example here?

Perhaps something like:
```
For example, if you have a server called example.com which stored git repos under /var/git/repos/name-of-repo.git and allowed access over ssh using port 2222 with the git username, you can configure earthly to recognize it as example.com/name-of-repo by configuring it with:

    config git "{example: {pattern: 'example.com/([^/]+)', substitute: 'ssh://git@example.com:2222/var/git/repos/\$1.git', auth: ssh}}"
    
 ```

Reply:
100%. I'll just use this?

Reply:
go for it :)